pkg/storegateway/indexheader/stream_binary_reader.go

```diff
@@ -148,7 +175,107 @@ func newFileStreamBinaryReader(path string, postingOffsetsInMemSampling int, log
 		return nil, err
 	}
 
-	return r, nil
+	return r, err
+}
+
+// loadSparseFromDisk loads in sparse index-header from disk.
+func (r *StreamBinaryReader) loadSparseFromDisk(ctx context.Context, logger log.Logger, sparseHeadersPath string, sparseData []byte, postingOffsetsInMemSampling int) (err error) {
+	span, _ := tracing.StartSpan(ctx, "index_header_load_sparse_header")
+	defer span.Finish()
+
+	start := time.Now()
+	defer level.Info(logger).Log("msg", "loaded sparse index-header from disk", "path", sparseHeadersPath, "elapsed", time.Since(start))
+
+	level.Debug(logger).Log("msg", "reading from sparse index-header file", "path", sparseHeadersPath)
+
+	sparseHeaders := &indexheaderpb.Sparse{}
+
+	gzipped := bytes.NewReader(sparseData)
+	gzipReader, err := gzip.NewReader(gzipped)
+	if err != nil {
+		return fmt.Errorf("failed to create sparse index-header reader: %w", err)
+	}
+
+	sparseData, err = io.ReadAll(gzipReader)
+	if err != nil {
+		return fmt.Errorf("failed to read sparse index-header: %w", err)
+	}
+
+	if err := sparseHeaders.Unmarshal(sparseData); err != nil {
+		return fmt.Errorf("failed to decode sparse index-header file: %w", err)
+	}
+
+	r.symbols, err = streamindex.NewSymbolsFromSparseHeader(r.factory, sparseHeaders.Symbols, r.indexVersion, int(r.toc.Symbols))
+	if err != nil {
+		return fmt.Errorf("cannot load symbols from sparse index-header: %w", err)
+	}
+
+	r.postingsOffsetTable, err = streamindex.NewPostingOffsetTableFromSparseHeader(r.factory, sparseHeaders.PostingsOffsetTable, int(r.toc.PostingsOffsetTable), postingOffsetsInMemSampling)
+	if err != nil {
+		return fmt.Errorf("cannot load postings offset table from sparse index-header: %w", err)
+	}
+
+	return nil
+}
+
+// constructSparseFromIndexHeader loads in symbols and postings offset table from the index-header and stores a sparse version.
```

Comment:
We're loading the index-header here, we're not "storing" anything on disk. Please update the comment and change the name of the span.